src/ui/public/inspector/build_tabular_inspector_data.js

```diff
@@ -17,41 +17,51 @@
  * under the License.
  */
 
+import { set } from 'lodash';
+import { createFilter } from '../vis/vis_filters';
 import { FormattedData } from './adapters/data';
 
 /**
  * This function builds tabular data from the response and attaches it to the
  * inspector. It will only be called when the data view in the inspector is opened.
  */
 export async function buildTabularInspectorData(table, queryFilter) {
-  const columns = table.columns.map((col, index) => {
+  const rows = table.rows.map(row => {
+    return table.columns.reduce((prev, cur, colIndex) => {
+      const value = row[cur.id];
+      const fieldFormatter = cur.aggConfig.fieldFormatter('text');
+      prev[`col-${colIndex}-${cur.aggConfig.id}`] = new FormattedData(value, fieldFormatter(value));
+      return prev;
+    }, {});
+  });
+
+  const columns = table.columns.map((col, colIndex) => {
     const field = col.aggConfig.getField();
     const isCellContentFilterable =
       col.aggConfig.isFilterable()
       && (!field || field.filterable);
     return ({
       name: col.name,
-      field: `col${index}`,
-      filter: isCellContentFilterable && ((value) => {
-        const filter = col.aggConfig.createFilter(value.raw);
+      field: `col-${colIndex}-${col.aggConfig.id}`,
+      filter: isCellContentFilterable && (value => {
+        const rowIndex = rows.findIndex(row => row[`col-${colIndex}-${col.aggConfig.id}`].raw === value.raw);
+        const filter = createFilter(table, colIndex, rowIndex, value.raw);
         queryFilter.addFilters(filter);
       }),
-      filterOut: isCellContentFilterable && ((value) => {
-        const filter = col.aggConfig.createFilter(value.raw);
-        filter.meta = filter.meta || {};
-        filter.meta.negate = true;
+      filterOut: isCellContentFilterable && (value => {
+        const rowIndex = rows.findIndex(row => row[`col-${colIndex}-${col.aggConfig.id}`].raw === value.raw);
+        const filter = createFilter(table, colIndex, rowIndex, value.raw);
+        const notOther = value.raw !== '__other__';
+        const notMissing = value.raw !== '__missing__';
+        if (Array.isArray(filter)) {
+          filter.forEach(f => set(f, 'meta.negate', (notOther && notMissing)));
+        } else {
+          set(filter, 'meta.negate', (notOther && notMissing));
```

Comment:
@ppisljar Any concerns about this treatment for `meta.negate`? It somewhat duplicates `vis_filters.filter`: https://github.com/elastic/kibana/blob/9912434f6586259a2129f43c3999bb2874292bf3/src/ui/public/vis/vis_filters.js#L84-L108

I looked at `filter` to see if it could be pulled out and shared in both places, but it feels like the `event` object would make this a bit hard to accomplishj, as inspector is only taking in tabified data anyway.